src/main/java/nl/tudelft/contextproject/model/level/Room.java

```diff
@@ -0,0 +1,201 @@
+	 */
+	public Room(String folder) {
+		this.folder = folder;
+		entities = new HashSet<Entity>();
+		lights = new ArrayList<Light>();
+		try {
+			size = getSizeFromFileName(folder);
+			mazeTiles = new MazeTile[size.getWidth()][size.getHeight()];
+			RoomParser.importFile(folder, mazeTiles, entities, lights, 0, 0);	
+			setDoors();
+		} catch (IOException e) {
+			Logger.getLogger("MazeGeneration").severe("Unable to correctly read name!");
+		}
+	}
+
+	/**
+	 * Get size of room from folder name.
+	 * @param folder
+	 * 		folder of the Room
+	 * @return
+	 * 		size of room
+	 * @throws IOException 
+	 * 		if pattern of roomFolder does not match
+	 */
+	protected Size getSizeFromFileName(String folder) throws IOException {
+		int width = 0;
+		int height = 0;
+		String fileName = RoomParser.getMapFile(folder).getName();
+		Matcher m = PATTERN.matcher(fileName);
+		if (m.matches()) {
+			width = Integer.parseInt(m.group("width"));
+			height = Integer.parseInt(m.group("height"));
+			return size = new Size(width, height);
+		} else {
+			throw new IOException("Expected a .crf file present in the folder with the correct name");
+		}
+	}
+
+	/**
+	 * Set the entrance and exit doors of the room.
+	 * Needed during graph creation.
+	 */
+	public void setDoors() {
+		entranceDoorsLocations = new ArrayList<Vec2I>();
+		exitDoorLocations = new ArrayList<Vec2I>();
+
+		for (int i = 0; i < size.getWidth(); i++) {
+			for (int j = 0; j < size.getHeight(); j++) {
+				if (mazeTiles[i][j] != null) {
+					TileType mazeTile = mazeTiles[i][j].getTileType();
+					if (mazeTile == TileType.DOOR_ENTRANCE) {
+						entranceDoorsLocations.add(new Vec2I(i, j));
+					} else if (mazeTile == TileType.DOOR_EXIT) {
+						exitDoorLocations.add(new Vec2I(i, j));
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Set mazeTiles array.
+	 * @param mazeTiles
+	 * 		maze tiles to set
+	 */
+	public void setMazeTiles(MazeTile[][] mazeTiles) {
+		this.mazeTiles = mazeTiles;
+	}
+
+	/**
+	 * Simple hash code of tiles used for testing.
+	 */
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + Arrays.deepHashCode(mazeTiles);
+		return result;
+	}
+
+	/**
+	 * Simple equals used for testing.
+	 * Only the TileType[][] array has to be equal;
+	 */
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		Room other = (Room) obj;
+		if (!other.size.equals(size)) 
+			return false;
+
+		for (int i = 0; i < size.getWidth(); i++) {
+			for (int j = 0; j < size.getHeight(); j++) {
+				if (mazeTiles[i][j] == null && mazeTiles[i][j] == null) {
+					if (mazeTiles[i][j] == null) return true;
+					return false;
+				}
+				else if (mazeTiles[i][j].getTileType() != other.mazeTiles[i][j].getTileType()) {
+					return false;
+				}
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Set size for testing.
```

Comment:
please add a newline after this line, so the @ lines are seperated from the method description